File: Easy/121/121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let maxPro = 0;
  let minPri = prices[0];
  for (price of prices) {
    minPri = Math.min(minPri, price);
    maxPro = Math.max(maxPro, price - minPri);
  }
  return maxPro;
};

console.log("It's Showtime Folks!");
console.log(`Solution: ${maxProfit([7, 1, 5, 3, 6, 4])}`);

File: easy/121/121.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        maxPro = 0
        minPri = prices[0]
        for price in prices:
            minPri = min(minPri, price)
            maxPro = max(maxPro, price - minPri)
        return maxPro


print("It's Showtime Folks!")
print(f"Solution: {Solution().maxProfit([7, 1, 5, 3, 6, 4])}")
